### ISSUE TITLE
2.5.10 does not see 2.5.11 as update

### ISSUE THREAD

**Issue by rmehner** [PostgresApp/PostgresApp#686]
:wave:

When I check for updates on 2.5.10 I'd expect 2.5.11 to show up, but it tells me that I'm up to date. However, looking over at your releases, it looks like [2.5.11](https://github.com/PostgresApp/PostgresApp/releases/tag/v2.5.11) is available :)

Might be a duplicate of #536

This is absolutely not a blocker in any form, so please don't stress about it. It's more of a nudge that you might have forgotten to publish it to the update channels or something.

Thanks for all your work!

**Comment by jakob** [comment]
If you have Postgres.app with PG 14, then 2.5.10 is the latest version. The next update to PG 14 is scheduled for November.

The automatic update check only checks for updates for the same major version you have downloaded, since upgrading to a new major version can not be done automatically and requires manually upgrading the database (typically via dump/restore).

**Comment by jakob** [comment]
Oh, apologies, I just checked again to be sure and it looks like the multi-version bundle (with PG 10-14) does not show the update to the new multi-version bundle (with 10-15). I think you are right. Sorry.

**Comment by jakob** [comment]
OK, I thought some more about it, and I guess the reason why I didn't change the sparkle feed for the PG 10-14 bundle is that the PG 10-15 bundle contains no updates for PG 10-14. The only change is that it includes PG 15 as well.

So you could argue that users of the PG 10-14 bundle don't need the update, because there are no updates for their installed servers.

But if people download the bundle "with all supported PG versions", it kinda would make sense to install new PostgreSQL versions as soon as they are available. It sounds like that is what you expected to happen. (And I think it's also what I would expect to happen).

@tbussmann I don't remember if we talked about this, is it okay if I just update the sparkle feed?

**Comment by tbussmann** [comment]
We already have the 10-15 feed checked by the 10-15 bundle (started with the beta version). I was expecting to add a note regarding the possibility of a manual download of the 10-15 or 15 bundle in the next 10-14 and 14 updates of November. If you'd like to already offer an inline update to the 10-14 users, I wouldn't object that either. But in that case we should think about renaming the channel to something like `updates_all_current.xml` - maybe we should handle future betas with different [channels within one feed](https://sparkle-project.org/documentation/publishing/#channels)

What happens on these bundle changes:
* Adding a version to the bundle does only change the `latest` symlink for current users. Using newer clients against older servers isn't a problem - but linking against (and esp. compiling and installing extensions via PGXS) could introduce some confusion for exiting users of the bundle.
* The other action that changes a feed usually is the removal of no longer supported versions. As this is a potentially breaking change, we did this in the past by defining a new feed with a cross-feed update that has a special note not to perform it if using the to-be removed version.

The most elegant way would be to combine addition and removal of versions in only one change per year, but that way we would need to deliver the out of support versions for almost one year after their [final release](https://www.postgresql.org/support/versioning/) in Nov. (In [Feb., May, Aug.](https://www.postgresql.org/developer/roadmap/) till the new version is introduced in Sep./Oct.) 

In any case, we should warn users of the soon to be removed versions with the last two scheduled updates in Aug. and Nov. to plan their migrations.

**Comment by rmehner** [comment]
First of all: Thank you @jakob and @tbussmann for taking your time to reply in such detail, even for such a minor issue.

To chime in:

> But if people download the bundle "with all supported PG versions", it kinda would make sense to install new PostgreSQL versions as soon as they are available. It sounds like that is what you expected to happen. (And I think it's also what I would expect to happen).

This is exactly what I'd expect to happen. I have multiple apps with different Postgres versions and I'm slowly upgrading them to Postgres 15 as time allows. That's why I rely on having the ability to run different Postgres versions locally thanks to Postgres.app. I'd expect to be offered an update to 2.5.11 that would allow me to start a Postgres 15 server, while keeping the ability to run older versions. I don't expect my data to be upgraded / migrated.

**Comment by jakob** [comment]
Sorry for not responding, I kind of forgot about this issue for a week. In any case, I have decided to update the sparkle feed for 10-14 to now offer the 10-15 update, because I really think that people who download a bundle "with all supported versions" probably expect to have all supported versions available.

I also realised we've already done something like this before, the 10-12 bundle already redirects to the 10-14 bundle.

In my opinion, the only potential problem is the changed "latest" symlink, but I can't think of a scenario where it would cause a bug. When linking against any libraries, the linker should use the absolute paths embedded in the libraries or from pg_config, so the symlink change shouldn't affect previously built things.
